lib/countries/data.rb

```diff
@@ -81,8 +84,23 @@ def sync_translations!
 
       private
 
+      def synchronized(&block)
+        if use_mutex?
+          @@mutex.synchronize(&block)
+        else
+          block.call
+        end
+      end
+
+      def use_mutex?
+        # Stubbed in testing
+        true
+      end
+
       def load_required?
-        @@cache.empty?
+        synchronized do
+          @@cache.empty?
```

Comment:
Why an independent `synchronized` block for this method? Why not put inside here?:

``` ruby
        synchronized do
          @@cache = load_cache %w(countries.json)
          @@_country_codes = @@cache.keys
          @@cache = @@cache.merge(@@registered_data)
          @@cache
        end
```